src/server/game/Entities/Object/Object.cpp

```diff
@@ -2305,13 +2305,91 @@ void WorldObject::MovePosition(Position &pos, float dist, float angle)
     pos.SetOrientation(GetOrientation());
 }
 
+// Only call this function when you already checked for collision with vmap/gameobject.
+// You should check for collision again after this one has been called.
+// excludeCollisionHeight should only be true if you had collision, it wont add it to raycasts for dest position.
+float WorldObject::SelectBestZForDestination(float x, float y, float z, bool excludeCollisionHeight) const
+{
+    if (Unit const* unit = ToUnit())
+        if (unit->IsFlying()) // Also check if in air ?
```

Comment:
if its not expensive, it should

also, skipping any correction for flying units feels scary

Reply:
It will only skip terrain because I didn't find any way to check it :/

I could just take the isInAir check that's used somewhere in updatemovementflags iirc and use that one for the comment tho